daily_milestone.go

```diff
@@ -158,34 +177,42 @@ func getMilestones(baseURL string, token string, projectID string) ([]simpleMile
 func createMilestoneData(advance int, timeInterval string) []simpleMilestone {
 	today := time.Now().Local()
 	list := []simpleMilestone{}
-	for i := 0; i < advance; i++ {
-		if timeInterval == "daily" {
+
+	switch {
+	case timeInterval == "daily":
+		for i := 0; i < advance; i++ {
 			date := today.AddDate(0, 0, i).Format("2006-01-02")
 			milestone := simpleMilestone{}
 			milestone.Title = date
 			milestone.DueDate = date
 			list = append(list, milestone)
-		} else if timeInterval == "weekly" {
-			today = today.AddDate(0, 0, 7)
-			year, week := today.ISOWeek()
-			milestoneYear := strconv.Itoa(year)
-			milestoneWeek := strconv.Itoa(week)
+		}
+	case timeInterval == "weekly":
+		year, week, lastDay := lastDayWeek(today)
+
+		for i := 0; i < advance; i++ {
+
 			milestone := simpleMilestone{}
-			milestone.Title = milestoneYear + "-w" + milestoneWeek
-			milestone.DueDate = today.Format("2006-01-02")
+			milestone.Title = strconv.Itoa(year) + "-w" + strconv.Itoa(week)
+			milestone.DueDate = lastDay.Format("2006-01-02")
+			year, week, lastDay = lastDayWeek(lastDay)
```

Comment:
Line 198 seems like it wouldn't affect anything.

Reply:
It's iterating through the weeks

Reply:
Ahh you are iterating for the next iteration in the last line. Feels like a strange flow, but we can leave that for now.